lib/connection.c

```diff
@@ -420,13 +425,30 @@ sf_bool STDCALL curl_post_call(SF_CONNECT *sf,
                                     SF_SQLSTATE_UNABLE_TO_CONNECT);
                 break;
             }
+
+            if(counter_for_code == 1 && sf->log_spec_info){
+                cJSON *newJson = snowflake_cJSON_Duplicate(*json, cJSON_True);
+                const char* del = "rowset";
+                //delete the sensitive information in case it leaks to customer
+                snowflake_cJSON_DeleteItemFromObject(newJson, del, cJSON_True);
+                log_info("Query continue in progress, query information: %s", snowflake_cJSON_Print(newJson));
+                //free the memory
+                snowflake_cJSON_free(newJson);
+                counter_for_code = 0;
+            }
+            counter_for_code++;
         }
 
         if (stop) {
             break;
         }
 
         ret = SF_BOOLEAN_TRUE;
+        counter++;
+        if(counter == 1000){
```

Comment:
Let's add the param check here as well (similar to what I suggested for previous one)